<?xml version="1.0" ?>
<launch>

  <arg name="config"		default="aist"/>
  <arg name="scene"		default="$(arg config)_base_scene"/>
  <arg name="sim"		default="false"/>
  <arg name="rviz_config"       default=""/>
  <arg name="debug"		default="false"/>

  <param name="use_real_robot"	value="$(eval not arg('sim'))"/>
  <include file="$(find aist_routines)/launch/load_config.launch">
    <arg name="config"	value="$(arg config)"/>
  </include>

  <!-- Launch gazebo if "sim == true". -->
  <include if="$(arg sim)"
  	   file="$(find aist_gazebo)/launch/$(arg config)_gazebo.launch">
    <arg name="scene"	value="$(arg scene)"/>
  </include>

  <!-- Load robot urdf to param "robot_description", spawn
       robot_state_publisher and launch real robots if "sim" ==
       "false". -->
  <group unless="$(arg sim)">
    <param name="robot_description"
  	   command="$(find xacro)/xacro --inorder '$(find
  		    aist_scene_description)/urdf/$(arg scene).urdf.xacro'"/>
    <node name="robot_state_publisher"
  	  pkg="robot_state_publisher" type="robot_state_publisher"/>
    <include file="$(find aist_routines)/launch/connect_$(arg
		   config)_robots.launch"/>
  </group>

  <!-- Launch planner -->
  <include file="$(find aist_routines)/launch/$(arg config)_moveit_planning_execution.launch">
    <arg name="sim"		value="$(arg sim)"/>
    <arg name="rviz_config"	value="$(arg rviz_config)"/>
    <arg name="debug"		value="$(arg debug)"/>
  </include>

</launch>
